### ISSUE TITLE
nix 2.7.0 installed on centos 7 `nix profile install <installable>` fails for certain `<installable>`

### ISSUE THREAD

**Issue by typetetris** [NixOS/nix#6373]
**Bug Description**

1. Take a freshly installed centos 7 (I used virtualbox) with a sudoer user added.
2. Install nix via `sh <(curl -L https://nixos.org/nix/install) --daemon`
3. Relogin
4. Add `experimental-features = nix-command flakes` to `/etc/nix/nix.conf`
5. `systemctl restart nix-daemon`
6. then
```
[centos@centos7-reproducer ~]$ nix profile install github:nixos/nixpkgs/nixos-21.11#jq
error: opening directory '/nix/store/0bzbsw9k557q3fcmac556caj96lnrpvs-jq-1.6': No such file or directory
[centos@centos7-reproducer ~]$ nix repl
Welcome to Nix 2.7.0. Type :? for help.

nix-repl> :lf github:nixos/nixpkgs/nixos-21.11
Added 14 variables.

nix-repl> p = legacyPackages.x86_64-linux

nix-repl> :b p.jq

This derivation produced the following outputs:
  bin -> /nix/store/x2pb4n6smfrig9sbcdj0j7zrw7h84dhb-jq-1.6-bin
  dev -> /nix/store/g8akk8w6vl6cagwpr3rq8qnlh5shfqdp-jq-1.6-dev
  doc -> /nix/store/1i18w8jp2a5vfz18a9z8wbrw5anydz48-jq-1.6-doc
  lib -> /nix/store/9vy1gay5qaniqz3bf93llwr5hsimx2hb-jq-1.6-lib
  man -> /nix/store/fvz9xr7yrgn0zbl8fcx4vay7xm8abzi8-jq-1.6-man
  out -> /nix/store/0bzbsw9k557q3fcmac556caj96lnrpvs-jq-1.6
[4 copied (0.0 MiB), 0.0 MiB DL]
nix-repl> :quit
[centos@centos7-reproducer ~]$ nix profile install github:nixos/nixpkgs/nixos-21.11#jq
[centos@centos7-reproducer ~]$
```

So trying to install `github:nixos/nixpkgs/nixos-21.11#jq` into the users profile right away
fails, but building it first in `nix repl` makes the installation in the users profile succeed afterwards.

Some other installables work, some others don't either.

**Expected behavior**

I expected `nix profile install <installable>` to either work or produce a build failure of some sort.

**`nix-env --version` output**
```
[centos@centos7-reproducer ~]$ nix-env --version
nix-env (Nix) 2.7.0
```

**Comment by edolstra** [comment]
I think this is fixed by #6328.

**Comment by typetetris** [comment]
Confirmed. Works with current master.
Would be really cool if a 2.7.1 release could be made with the bug fix!
